src/core/etl/src/Flow/ETL/Function/HTMLQuerySelector.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Function;
+
+use Dom\{Element, HTMLDocument};
+use Flow\ETL\Exception\RequiredPHPVersionException;
+use Flow\ETL\Row;
+
+final class HTMLQuerySelector extends ScalarFunctionChain
+{
+    public function __construct(
+        private readonly mixed $value,
+        private readonly ScalarFunction|string $selector,
+    ) {
+    }
+
+    public function eval(Row $row) : ?Element
+    {
+        if (!\class_exists('\Dom\HTMLDocument')) {
```

Comment:
could you actually move that to constructor? This way the exception is going to be thrown when you just use that function, otherwise it will crash only when pipeline is evaluated and only if it goes into this code.